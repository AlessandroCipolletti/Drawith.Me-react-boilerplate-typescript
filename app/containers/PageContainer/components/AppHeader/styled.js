import styled from 'styled-components'

import headerBG from 'images/UI/PageContainer/headerBG.png'
import logo from 'images/UI/PageContainer/logo.png'

import HEADER_HEIGHT from '../../constants'

export const Wrapper = styled.div`
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
  height: ${HEADER_HEIGHT};
  border-bottom: solid 1px ${props => props.theme.palette.gray[6]};
  background-image: url(${logo}), url(${headerBG});
  background-size: auto 90%, contain;
  background-repeat: no-repeat, repeat-x;
  background-position: center, 0% 0%;
  z-index: 1000;
`

export const HeaderButton = styled.div`
  top: 0px;
  width: ${HEADER_HEIGHT};
  height: ${HEADER_HEIGHT};
  float: ${props => props.float || 'left'};
  box-sizing: border-box;
  padding: ${props => props.theme.spacing.size1};
  color: ${props => props.theme.palette.white};
  font-size: ${props => props.theme.fontSizing.size6};
  line-height: ${HEADER_HEIGHT};
  cursor: pointer;
`
