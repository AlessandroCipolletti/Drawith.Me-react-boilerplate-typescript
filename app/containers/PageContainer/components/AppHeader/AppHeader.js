import React from 'react'
import PropTypes from 'prop-types'
import Theme from 'common/Theme'

import Icon from 'common/components/Icon'
import { Wrapper, HeaderButton } from './styled'
import { InfoButton } from './components'

const AppHeader = ({
  headerButtons = [], showInfoPopup,
}) => (
  <Wrapper>
    {headerButtons.map(button => (
      <HeaderButton
        key={button.id}
        float={button.float}
        onClick={button.action}
        className='HeaderButton'
      >
        {button.icon && <Icon
          name={button.icon}
          size={Theme.spacing.size6}
        />}
        {button.text || ''}
      </HeaderButton>
    ))}
    <InfoButton action={showInfoPopup} />
  </Wrapper>
)

AppHeader.propTypes = {
  headerButtons: PropTypes.array,
  showInfoPopup: PropTypes.func.isRequired,
}

export default AppHeader
