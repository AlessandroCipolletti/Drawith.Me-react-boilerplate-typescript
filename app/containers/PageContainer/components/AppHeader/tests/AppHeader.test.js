import React from 'react'
import { render } from 'react-testing-library'

// Import de Theme Provider
import { ThemeProvider } from 'styled-components'
import Theme from 'common/Theme'

import AppHeader from '../AppHeader'

describe('<AppHeader />', () => {

  const renderComponent = (props = {}) => (
    render(
      <ThemeProvider theme={Theme}>
        <AppHeader {...props} />
      </ThemeProvider>,
    )
  )

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    const showInfoPopup = () => {}

    renderComponent({ showInfoPopup })

    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect log an error in console if called without action prop', () => {
    const spy = jest.spyOn(global.console, 'error')

    renderComponent()

    expect(spy).toHaveBeenCalledTimes(2)
  })

  it('Should render and match the snapshot with default props', () => {
    const showInfoPopup = () => {}

    const {
      container: { firstChild },
    } = renderComponent({ showInfoPopup })

    expect(firstChild).toMatchSnapshot()
  })

  it('Should render and match the snapshot with default props', () => {
    const showInfoPopup = () => {}
    const headerButtons = [{
      id: '0',
      icon: 'info',
      action: ()=>{},
    }, {
      id: '1',
      text: 'text',
      action: ()=>{},
    }]

    const { container } = renderComponent({ showInfoPopup, headerButtons })
    const buttonsLength = container.querySelectorAll('.HeaderButton').length

    expect(buttonsLength).toBe(headerButtons.length)
  })

})
