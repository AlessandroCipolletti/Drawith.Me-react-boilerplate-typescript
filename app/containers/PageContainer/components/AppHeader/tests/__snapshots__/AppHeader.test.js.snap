// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader /> Should render and match the snapshot with default props 1`] = `
<div
  class="styled__Wrapper-sc-1a6eih8-0 jXWNUQ"
>
  <div
    class="IconButton__Wrapper-sc-15xi3uz-0 fctjIu"
    data-testid="IconButton"
  >
    <div
      class="Icon__Wrapper-xawg7a-0 hKOtws"
      color="#ffffff"
      role="button"
      size="32"
      tabindex="0"
    />
  </div>
</div>
`;
